src/crimson/osd/pg.cc

```diff
@@ -1771,4 +1771,21 @@ PG::already_complete(const osd_reqid_t& reqid)
   }
 }
 
+void PG::PGLogEntryHandler::remove(const hobject_t &soid) {
+  auto &backend = pg->get_backend();
+  backend.remove(soid, *t);
+  auto &osdriver = pg->get_osdriver();
+  OSDriver::OSTransaction _t(osdriver.get_transaction(t));
+  if (soid.snap < CEPH_MAXSNAP) {
+    auto &snap_mapper = pg->get_snap_mapper();
+    int r = snap_mapper.remove_oid(
+      soid,
+      &_t);
+    if (!(r == 0 || r == -ENOENT)) {
+      logger().debug("{}: remove_oid returned {}", __func__, cpp_strerror(r));
+      ceph_abort();
+    }
+  }
+}
```

Comment:
Can we please move this into `PG::clear_object_snap_mapping`?

Reply:
I think we need `remove_snap_mapped_object`. `clear_object_snap_mapping` is always used with `ceph::os::Transaction::remove`, which can be replace by `remove_snap_mapped_object`. What do you think?

Reply:
All coments addressed, please take a look again:-)